src/Frontend/src/views/throughputreport/setup/MasksView.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { makeDriverForTests, userEvent, render, screen } from "@component-test-utils";
+import MasksView from "@/views/throughputreport/setup/MasksView.vue";
+import { describe, expect, test } from "vitest";
+import * as precondition from "../../../../test/preconditions";
+import { useServiceControl } from "@/composables/serviceServiceControl";
+import { useServiceControlUrls } from "@/composables/serviceServiceControlUrls";
+import { minimumSCVersionForThroughput } from "@/views/throughputreport/isThroughputSupported";
+import flushPromises from "flush-promises";
+import Toast from "vue-toastification";
+import { disableMonitoring } from "../../../../test/drivers/vitest/setup";
+
+describe("MaskView tests", () => {
+  const serviceControlInstanceUrl = window.defaultConfig.service_control_url;
+
+  async function setup() {
+    const driver = makeDriverForTests();
+
+    disableMonitoring();
+
+    await driver.setUp(precondition.hasServiceControlMainInstance(minimumSCVersionForThroughput));
+    await driver.setUp(precondition.hasUpToDateServicePulse);
+    await driver.setUp(precondition.hasUpToDateServiceControl);
+    await driver.setUp(precondition.errorsDefaultHandler);
+
+    return driver;
+  }
+
+  async function renderComponent(body: string[] = []) {
+    const driver = await setup();
+    driver.mockEndpoint(`${serviceControlInstanceUrl}licensing/settings/masks`, { body });
+    useServiceControlUrls();
+    await useServiceControl();
```

Comment:
Directly calling implementation details from the system under test (SUT) indicates that the SUT is not written with testability in mind. 

One idea to fix this is to convert these two highlighted composables into Pinia stores. That way, all the data the SUT expects can be set in the arrangement section of the test via the Pinia testing API. Can the scope of the work you are doing be increased to also fix this? I suppose not.


Given that the SUT needs to be refactored with testability in mind, the current state of the code makes Acceptance Tests the best option to go with and leave component testing only for [presentational components](https://medium.com/prod-io/presentational-container-components-when-to-use-what-99be8fa28953) (components that receive the data through props). Testing components that are responsible for fetching the data (views) require us (today) to have these duplicate production calls in the testing setup (implementation details leakage), again, because the SUT is not test-friendly enough yet.

There is this [issue](https://github.com/Particular/ServicePulse/issues/1905) with the suggestion to refactor more of these chucky composables into Pinia store when applicable. useServiceControlUrls(), useServiceControl() and useLicense() are great candidates for that.

Consider adding a comment to that issue so that when the refactor happens, these tests can get updated to no longer require these calls.

Reply:
It is not ideal to deal with the environmental state on all the tests, so hiding this somewhere can be advantageous. Whether that is a store, a function or something else is debatable.
I don't think that if a component does data load, we need to use acceptance tests only.  (behind the scenes, something else is doing this heavy lifting to make it work anyway)
In my opinion, we should always follow the testing pyramid, where the heavier testing practices are the smallest set.

I am glad I persisted with component testing and got it working.